File: nicholas_christiansen/week-01/01-js-calculator and strings/js/warmup.js
console.log('WARMUP -----');

// What number's bigger?
console.log("----- raindrops -----");

/*
Write a Javascript program that will take a number (eg 28 or 1755 or 9, etc)
and output the following:
- If the number contains 3 as a factor, output 'Pling'.
- If the number contains 5 as a factor, output 'Plang'.
- If the number contains 7 as a factor, output 'Plong'.
- If the number does not contain 3, 5, or 7 as a factor, output the number as
  a string.
*/

const raindrops = function(num) {
  let text = '';
  if (num % 3 === 0) { text += 'Pling'; }
  if (num % 5 === 0) { text += 'Plang'; }
  if (num % 7 === 0) { text += 'Plong'; }
  if (text.length === 0) { text = num.toString(); }

  return text
}

console.log("---------- examples -----");

console.log(raindrops(28));
console.log(raindrops(1755));
console.log(raindrops(34));

console.log("---------- 0:99 -----");

for (var i = 0; i < 100; i++) {
  console.log(raindrops(i));
}

File: warmups/week04/day02_ruby_raindrops/warmup.rb
def raindrops num
  
  # create an empty string that will be apppended to
  string = ''

  # if statement to test if number if divisible by 3, and if so, print out 'Pling' (add Pling to the str)
  # if num % 3 == 0
  #   string += "Pling"
  # end

  # short conditional 'if' statements can be condensed down to a single line.
  string += "Pling" if num % 3 == 0

  # if num % 5 == 0
  #   string += "Plang"
  # end
  string += "Plang" if num % 5 == 0

  # if num % 7 == 0
  #   string += "Plong"
  # end
  string += "Plong" if num % 7 == 0

  # if string.length == 0
  #   string = num.to_s
  # end

  # If str is empty, ie not a factor of 3,5,7 then return num (as string)
  string = num.to_s if string.empty?

  # Ruby allows implicit return. Placing this value last in my program is the same as saying return str.
  string

end

puts raindrops 28
puts raindrops 34
puts raindrops 105
puts raindrops 1755
